pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -140,25 +165,48 @@ func (s *akvKMProvider) GetCertificates(ctx context.Context) (map[keymanagementp
 	for _, keyVaultCert := range s.certificates {
 		logger.GetLogger(ctx, logOpt).Debugf("fetching secret from key vault, certName %v,  keyvault %v", keyVaultCert.Name, s.vaultURI)
 
-		// fetch the object from Key Vault
-		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
 		startTime := time.Now()
+
+		// GetSecret is required so we can fetch the entire cert chain. See issue https://github.com/ratify-project/ratify/issues/695 for details
 		secretBundle, err := s.kvClient.GetSecret(ctx, s.vaultURI, keyVaultCert.Name, keyVaultCert.Version)
 		if err != nil {
+			// certificate is disabled, remove it from the map
+			var de autorest.DetailedError
+			if errors.As(err, &de) {
+				var re *azure.RequestError
+				if errors.As(de.Original, &re) {
+					if re.ServiceError.Code == "SecretDisabled" {
```

Comment:
I would define a method to check if it's a `secretDisabled` error, otherwise we have too many nested if statements.

Reply:
Method or function? If a method, are you thinking it should be on the client or the akvKMProvider?

Reply:
sorry for misleading, funtion should be enough

Reply:
Not a problem, I use those interchangeably also but thought I'd clarify.